lib/schema/schema.rb

```diff
@@ -62,11 +62,17 @@ def attribute_macro(attribute_name, type=nil, strict: nil, default: nil)
         end
       end
 
-      initialize_value = nil
-      if default.is_a? Proc
+      case default
+      when NilClass
+        initialize_value = nil
+      when Proc
         initialize_value = default
-      elsif !default.nil?
+      when String
         initialize_value = proc { default.clone }
+      when Numeric, TrueClass, FalseClass, Symbol
+        initialize_value = proc { default }
+      else
+        raise Schema::Attribute::Error, "The \"#{attribute_name}\" attribute is declared with a default value that is an object reference"
```

Comment:
In the future, change to: "Default values must be callable, like procs, lambdas, or objects that respond to the call method (Attribute: #{attribute_name})"